casper/src/main/scala/coop/rchain/casper/MultiParentCasper.scala

```diff
@@ -51,6 +51,10 @@ object MultiParentCasper {
       parentHashes: Set[BlockHash]
   ): F[ParentsMergedState] =
     for {
+      _ <- FatalError(
+            "Parents must not be empty to calculate pre-state. Genesis block pre-state is loaded from config."
+          ).raiseError.whenA(parentHashes.isEmpty)
+
```

Comment:
Why can't `parentHashes` be empty? Because the block must be based on other blocks? What about genesis?

Reply:
This function calculates state based on parent blocks (justifications) so without that state is mostly defined by configuration or default values. To make it more clear better to not mix this function with these values.